core-lib/Benchmarks/LeeSTM.ns

```diff
@@ -0,0 +1,603 @@
+(* *)
+class LeeSTM usingPlatform: platform andHarness: harness = (
+| private Benchmark = harness Benchmark.
+  private Array     = platform kernel Array.
+  private Vector    = platform kernel Vector.
+  private Mutex     = platform threading Mutex.
+  private Thread    = platform threading Thread.
+  private Transaction = platform transactions Transaction.
+  private system    = platform system.
+
+  private EMPTY      = 0.
+  private TEMP_EMPTY = 10000.
+  private OCC        = 5120.
+  private VIA        = 6000.
+  private BVIA       = 6001.
+  private TRACK      = 8192.
+  private MAX_WEIGHT = 1.
+
+  private dx = { { -1. 1.  0. 0 }. {  0. 0. -1. 1 } }.
+  private dy = { {  0. 0. -1. 1 }. { -1. 1.  0. 0 } }.
+|) (
+  class Lee = Benchmark (
+  | private data ::= nil.
+    private gridSize ::= 0.
+  |)(
+    public innerBenchmarkLoop: problemSize numThreads: numThreads = (
+      | totalLaidTracks lr threads m |
+      m:: Mutex new.
+      data isNil ifTrue: [ loadData: problemSize ].
+
+      totalLaidTracks:: 0.
+      lr:: LeeRouter new: data gridSize: gridSize.
+
+      threads:: Array new: numThreads withAll: [
+        Thread spawn: [ | t |
+          t:: LeeThread new: lr.
+          t run.
+          m critical: [
```

Comment:
@daumayr hm, looks like you're still using a mutex here. Is this on purpose?

Reply:
no , i thought i used a search function to get rid of all of them. apparently i missed one.

Reply:
fixed